front/src/Phaser/Game/GameMap.ts

```diff
@@ -79,7 +79,9 @@ export class GameMap {
         this.flatLayers = flattenGroupLayersMap(map);
         this.tiledObjects = this.getObjectsFromLayers(this.flatLayers);
         // NOTE: We leave "zone" for legacy reasons
-        this.areas = this.tiledObjects.filter((object) => ["zone", "area"].includes(object.type));
+        this.tiledObjects
+            .filter((object) => ["zone", "area"].includes(object.type))
+            .forEach((area) => this.areas.set(area.name, area));
```

Comment:
We could maybe issue a warning if 2 areas have the same name? (except in the case of an "empty" name that we should probably allow)

So maybe try something like:

```
if name empty => don't add anything to this.areas
if name not empty AND if the name is already used, put a warning in the console
```